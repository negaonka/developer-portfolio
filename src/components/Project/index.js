import React from 'react';
import github from '../../assets/github.png'

function Project({developer}){
    return(
    <div >
        <div className="container">
            <img alt='developer' src={developer.image} width="500px" height="300px"/> 
            <div className="middle">
                <a href={developer.github} target="_blank" rel="noreferrer">  
                    <img alt='github' src={github} width="40px" height="40px"/>
                </a>
                <a href={developer.application} target="_blank" rel="noreferrer" >  
                   <p>{developer.name}</p>
                </a>
            </div>
        </div>
    </div>
    )
}

export default Project;