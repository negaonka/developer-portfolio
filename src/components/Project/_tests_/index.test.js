import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Project from '..';

afterEach(cleanup);  //to clean all the memory data

const developer = [{
    image: "developer1",
    application:"https://api.github.com/users/mojombo",
    github:"https://github.com/mojombo",
    name: "Plantonic Game"
    }
  ]

describe('Project component', () => {
    // baseline test
    it('renders', () => {
        render(<Project developer={developer}/>);
      });
   
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Project developer={developer}/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
      });

    })