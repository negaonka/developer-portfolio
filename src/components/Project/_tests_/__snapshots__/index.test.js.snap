// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <img
        alt="developer"
        height="300px"
        width="500px"
      />
      <div
        class="middle"
      >
        <a
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="github"
            height="40px"
            src="github.png"
            width="40px"
          />
        </a>
        <a
          rel="noreferrer"
          target="_blank"
        >
          <p />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
